docs/user-guide/grpc.md

```diff
@@ -4,133 +4,209 @@
 
-## Example
+## Writing creating a gRPC service for Ambassador
 
-This tutorial assumes you have already followed the [Ambassador Getting Started](/user-guide/getting-started.html) guide. If you haven't done that already, you should do that now.
+There are many examples and walkthroughs on how to write gRPC applications so that is not what this article will aim to accomplish. If you do not yet have a service written you can find examples of gRPC services in all supported languages here: [gRPC Quickstart](https://grpc.io/docs/quickstart/)
 
-After completing [Getting Started](/user-guide/getting-started.html), you'll have a Kubernetes cluster running Ambassador and the Quote of the Moment service. Let's walk through adding the [Hello World gRPC service](https://github.com/grpc/grpc-go/tree/master/examples/helloworld) for this tutorial. 
+This document will use the [gRPC python helloworld example](https://github.com/grpc/grpc/tree/master/examples/python/helloworld) to demonstrate how to configure a gRPC service with Ambassador.
 
-## Mapping gRPC Services
+Follow the example up through [Run a gRPC application](https://grpc.io/docs/quickstart/python.html#run-a-grpc-application) to get started.
 
-Ambassador `Mapping`s are based on URL prefixes; for gRPC, the URL prefix is the full service name, including the package path. 
+### Dockerize
 
-For `Hello World`, in its [proto definition file](https://github.com/grpc/grpc-go/blob/master/examples/helloworld/helloworld/helloworld.proto), we see
+After building our gRPC application and testing it locally, we need to package it as a Docker container and deploy it to kubernetes.
+
+```shell
+$ touch Dockerfile
+```
+
+To run a gRPC application, we need to include the client/server and the protocol buffer definitions. 
+
+
+For gRPC with python, we need to install `grpcio` and the common protos. 
+
+```Dockerfile
+FROM python:2.7
+
+WORKDIR /grpc
+
+ENV PATH "$PATH:/grpc"
+
+COPY greeter_server.py /grpc
+COPY helloworld_pb2.py /grpc
+COPY helloworld_pb2_grpc.py /grpc
+
+RUN python -m pip install grpcio
+RUN python -m pip install grpcio-tools googleapis-common-protos
+
+CMD ["python", "./greeter_server.py"]
+
+EXPOSE 50051
+```
+
+Create the container and test it:
+
+```shell
+$ docker build -t <docker_hub>/grpc_example .
+$ docker run -p 50051:50051 <docker_hub>/example
+```
+Where `<docker_hub>` is your docker profile.
+
+Switch to another terminal and, from the same directory, run the `greeter_client`.
+The application should run exactly the same.
```

Comment:
I don't think you reference running it outside the container, so maybe just list the expected output?